async.zsh

```diff
@@ -178,6 +178,7 @@ _async_worker() {
 
 	local request do_eval=0
 	local -a cmd
+	local cat_bin=$(which -p cat)
```

Comment:
We could simplify this by using the built-in `command`, so essentially just `command cat` when invoking it.

```
       command [ -pvV ]
              The command word is taken to be the name of an external command,
              rather than a shell function or builtin.   If the POSIX_BUILTINS
              option  is  set, builtins will also be executed but certain spe-
              cial properties of them are suppressed. The  -p  flag  causes  a
              default  path  to be searched instead of that in $path. With the
              -v flag, command is similar to whence and with -V, it is equiva-
              lent to whence -v.
```